auth0/src/main/java/com/auth0/android/provider/LogoutManager.java

```diff
@@ -0,0 +1,87 @@
+package com.auth0.android.provider;
+
+import android.content.Context;
+import android.net.Uri;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+import android.support.annotation.VisibleForTesting;
+import android.util.Log;
+
+import com.auth0.android.Auth0;
+import com.auth0.android.Auth0Exception;
+import com.auth0.android.callback.BaseCallback;
+
+import java.util.HashMap;
+import java.util.Map;
+
+@SuppressWarnings("WeakerAccess")
+class LogoutManager extends ResumableManager {
+
+    private static final String TAG = LogoutManager.class.getSimpleName();
+
+    private static final String KEY_CLIENT_ID = "client_id";
+    private static final String KEY_TELEMETRY = "auth0Client";
+
+    private final Auth0 account;
+    private final BaseCallback<Void, Auth0Exception> callback;
+    private final Map<String, String> parameters;
+
+    private CustomTabsOptions ctOptions;
+
+    LogoutManager(@NonNull Auth0 account, @NonNull BaseCallback<Void, Auth0Exception> callback, @NonNull Map<String, String> parameters) {
+        this.account = account;
+        this.callback = callback;
+        this.parameters = new HashMap<>(parameters);
+    }
+
+    void setCustomTabsOptions(@Nullable CustomTabsOptions options) {
+        this.ctOptions = options;
+    }
+
+    void startLogout(Context context) {
+        addClientParameters(parameters);
+        Uri uri = buildLogoutUri();
+
+        AuthenticationActivity.authenticateUsingBrowser(context, uri, ctOptions);
+    }
+
+    @Override
+    boolean resume(AuthorizeResult result) {
+        if (result.isCanceled()) {
+            Auth0Exception exception = new Auth0Exception("The user closed the browser app and the log out was cancelled.");
+            callback.onFailure(exception);
+        } else {
+            callback.onSuccess(null);
```

Comment:
Feels a little weird to require the callback handler to implement success handler using `Void`. I believe typically the Void type is used for reflection purposes. What's the downside to adding a new Callback class - wouldn't it be pretty small and simple?

Reply:
Yes, it's used because of generics. That way a single callback class can handle different payload types and different exception types.

Yes, I can add a new class but I'm trying to use what we already have shipped. The only downside is another public class. The upside is not having to specify the types with the diamond symbol. However we come up with a good name for a callback that returns nothing on success and throws an Auth0Exception otherwise (our generic exception type) I don't mind adding it.